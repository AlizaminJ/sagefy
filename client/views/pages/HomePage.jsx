import React from 'react'
import { Link } from 'react-router-dom'
import { string } from 'prop-types'
import Icon from '../components/Icon'
import Footer from '../components/Footer'
import ExternalLink from '../components/ExternalLink'

export default function HomePage({ role }) {
  return (
    <div className="HomePage">
      {/* <section className="text-align-center">
        <h1>
          What do you want to learn? <Icon i="search" s="xxl" />
        </h1>
        <form action="/search-subjects">
          <p>
            <input
              type="search"
              size="40"
              placeholder="example: Music"
              autoFocus
              name="q"
            />
          </p>
          <p>
            <button type="submit">
              <Icon i="search" /> Search
            </button>
          </p>
        </form>
      </section> */}

      <section>
        <blockquote>
          <h1>
            Hey friends... <Icon i="friends" s="xxl" />
          </h1>

          <p>So I&apos;m rebuilding Sagefy to let anyone</p>
          <ol>
            <li>
              <em>learn</em> without an account,
            </li>
            <li>
              <em>make</em> content without an account,
            </li>
            <li>
              and to drastically <em>simplify</em> contributing.
            </li>
          </ol>
          <p>
            Sagefy is temporarily limited. But here&apos;s some things you can
            do now:
          </p>
          <ul>
            <li>
              <ExternalLink href="https://docs.sagefy.org/mocks">
                <Icon i="view" /> <strong>View</strong> the prototype
              </ExternalLink>{' '}
              of what&apos;s coming!
            </li>
            <li>
              <Link to="/sign-up">
                <Icon i="signUp" /> <strong>Sign up</strong>
              </Link>
              , and we&apos;ll let you know as we release new stuff.
            </li>
            <li>
              <ExternalLink href="https://sgfy.xyz/devupdates">
                <Icon i="updates" /> <strong>Subscribe</strong> to biweekly
                email updates
              </ExternalLink>
              .
            </li>
          </ul>
          <p>
            <small>Last updated 2019 Mar 8</small>
          </p>
        </blockquote>
      </section>

      {role === 'sg_anonymous' ? (
        <section className="text-align-right">
          <p>
            <small>
              <Link to="/log-in">
                <Icon i="logIn" /> Log In
              </Link>{' '}
              or{' '}
              <Link to="/sign-up">
                <Icon i="signUp" /> Sign Up
              </Link>
            </small>
          </p>
        </section>
      ) : (
        <section className="text-align-right">
          <p>
            <small>
              <Link to="/dashboard">
                Go to <Icon i="dashboard" /> Dashboard
              </Link>
            </small>
          </p>
        </section>
      )}

      {/* TODO make functional
      <section>
        <h2>
          ...or try something popular <Icon i="popular" s="xl" />
        </h2>
        <table>
          <tr>
            <td className="text-align-center collapse-margins">
              <p>
                <button type="button">
                  <Icon i="up" />
                </button>
              </p>
              <code>12</code>
            </td>
            <td className="collapse-margins">
              <h3>
                <Link to="/choose-next">
                  An Introduction to Electronic Music
                </Link>
              </h3>
              <p>
                A small taste of the basics of electronic music. Learn the
                concepts behind creating and modifying sounds in an electronic
                music system. Learn the ideas behind the tools and systems we
                use to create electronic music.
              </p>
            </td>
          </tr>
        </table>
      </section> */}

      <section>
        <div className="text-align-center collapse-margins">
          <img src="/astrolabe.svg" height="120" alt="astrolabe" />
          <h2>Sagefy</h2>
          <p>
            <em>Learn anything, adapted for you. Free.</em>
          </p>
        </div>
      </section>

      <section>
        <ul>
          <li>
            <Icon i="adapt" /> <strong>Adaptive Learning.</strong> Sagefy
            optimizes based on what you already know and what your goal is. Get
            the most out of your time and effort spent.
          </li>
          <li>
            <Icon i="open" /> <strong>Open Content.</strong> Anyone can view,
            share, create, and edit content. Because anyone can contribute, you
            can learn anything you want.
          </li>
          <li>
            <ExternalLink href="https://youtu.be/gFn4Q9tx7Qs">
              <Icon i="video" /> Watch the overview video
            </ExternalLink>
            .
          </li>
        </ul>
      </section>

      <Footer role={role} />
    </div>
  )
}

HomePage.propTypes = {
  role: string,
}

HomePage.defaultProps = {
  role: 'sg_anonymous',
}
