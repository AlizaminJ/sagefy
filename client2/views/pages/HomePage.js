import React from 'react'

import Icon from '../components/Icon'

// TODO fix links for SPA routing
// TODO open external in new window
// TOOD isLoggedIn prop

export function HomePageCta() {
  return (
    <a href="/sign_up" className="home__cta-button">
      <Icon name="sign-up" />
      {" Let's Learn"}
    </a>
  )
}

export function WrappedIcon({ name }) {
  return (
    <span className="home__icon-wrap">
      <Icon name={name} />
    </span>
  )
}

export function HomePageInfo() {
  return (
    <div>
      <section>
        <hgroup>
          <h2>What is Sagefy?</h2>
          <h5>Sagefy is an open-content adaptive learning platform.</h5>
        </hgroup>
        <p>
          <strong>Adaptive Learning.</strong>
          {
            ' Sagefy optimizes based on what you already know and what your goal is. Get the most out of your time and effort spent.'
          }
        </p>
        <p>
          <strong>Open-Content.</strong>
          {
            ' Anyone can view, share, create, and edit content. Because anyone can contribute, you can learn anything you want.'
          }
        </p>
        <HomePageCta />
      </section>
      <section>
        <hgroup>
          <h2>What is Sagefy?</h2>
          <h5>Sagefy is an open-content adaptive learning platform.</h5>
        </hgroup>
        <p>
          <strong>Adaptive Learning.</strong> Sagefy optimizes based on what
          you already know and what your goal is. Get the most out of your time
          and effort spent.
        </p>
        <p>
          <strong>Open-Content.</strong> Anyone can view, share, create, and
          edit content. Because anyone can contribute, you can learn anything
          you want.
        </p>
        <HomePageCta />
      </section>
      <section>
        <h2>How do I learn with Sagefy?</h2>
        <ol className="home__ul--how">
          <li>
            <img
              src="https://i.imgur.com/2QSMPNs.png"
              alt="sign up for account"
            />{' '}
            Create an account.
          </li>
          <li>
            <img
              src="https://i.imgur.com/xKRaoff.png"
              alt="search a subject"
            />{' '}
            Find and add a subject.
          </li>
          <li>
            <img src="https://i.imgur.com/MYTGawz.png" alt="choose unit" />{' '}
            Choose your unit.
          </li>
          <li>
            <img
              src="https://i.imgur.com/yjeVPiq.png"
              alt="multiple choice card"
            />{' '}
            Learn.
          </li>
        </ol>
        <HomePageCta />
      </section>
      <section>
        <h2>Popular Subjects</h2>
        <ul className="home__ul--popular-subjects">
          {/* TODO use preview--subject__head component */}
          <li>
            <div className="preview--subject__head">
              <h3>
                <i className="icon icon-subject" />{' '}
                <a href="/subjects/UIe3mx3UTQKHDG2zLyHI5w/landing">
                  An Introduction to Electronic Music
                </a>
              </h3>
              <p>
                A small taste of the basics of electronic music. Learn the
                concepts behind creating and modifying sounds in an electronic
                music system. Learn the ideas behind the tools and systems we
                use to create electronic music.
              </p>
            </div>
          </li>
        </ul>
        <HomePageCta />
      </section>
      <section>
        <h2>Why learn with Sagefy?</h2>
        <ul className="home__ul--why">
          <li>
            <WrappedIcon name="learn" />
            <span>
              <strong>Learn</strong> any subject.
            </span>
          </li>
          <li>
            <WrappedIcon name="fast" />
            <span>
              <strong>Skip</strong> what you already know.
            </span>
          </li>
          <li>
            <WrappedIcon name="grow" />
            <span>
              <strong>Build up</strong> to where you need to be.
            </span>
          </li>
          <li>
            <WrappedIcon name="search" />
            <span>
              <strong>Choose</strong> your own path.
            </span>
          </li>
          <li>
            <WrappedIcon name="learn" />
            <span>
              Learn <strong>deeply</strong>, instead of skimming the top.
            </span>
          </li>
          <li>
            <WrappedIcon name="follow" />
            <span>
              Stay <strong>motiviated</strong> with different types of cards.
            </span>
          </li>
          <li>
            <WrappedIcon name="good" />
            <span>
              Focus on what you want to learn with{' '}
              <strong>no distractions.</strong>
            </span>
          </li>
          <li>
            <WrappedIcon name="create" />
            <span>
              Create and edit <strong>any</strong> content.
            </span>
          </li>
          <li>
            <WrappedIcon name="topic" />
            <span>
              <strong>Discuss</strong> anything as you learn.
            </span>
          </li>
        </ul>
        <HomePageCta />
      </section>
      <section>
        <h2>What does Sagefy provide?</h2>
        <iframe
          width="560"
          height="315"
          title="What is Sagefy? Video"
          src="https://www.youtube.com/embed/gFn4Q9tx7Qs"
          frameBorder="0"
          allowFullScreen="true"
        />
        <p>
          Also check out the in-detail{' '}
          <a href="https://stories.sagefy.org/why-im-building-sagefy-731eb0ceceea">
            article on Medium
          </a>.
        </p>
        <HomePageCta />
      </section>
      <section>
        <h2>Comparison</h2>
        <ul>
          <li>
            <strong>Classroom</strong>: When we adapt the content to what you
            already know, we keep the motivation going and reduce effort and
            time. Classrooms are a difficult place to get personal. Sagefy
            optimizes for what you already know, every time.
          </li>
          <li>
            <strong>Learning Management Systems</strong>: Great cost and time
            savings come from using technology. LMSs are designed to support
            the classroom model. With Sagefy, you get both the benefits of
            online learning and a highly personalized experience.
          </li>
          <li>
            <strong>Closed Adaptive Systems</strong>: Pursue your own goals.
            Closed systems means only select topics are available. An
            open-content system like Sagefy reaches a range of topics.
          </li>
          <li>
            <strong>Massive Online Courses</strong>: MOOCs reach a large range,
            but offer little adaption and only support expert-created content.
            Sagefy has no deadlines -- learn when you see fit.
          </li>
          <li>
            <strong>Flash Cards</strong>: Flash cards are great for memorizing
            content. But what about integration and application of knowledge?
            Sagefy goes deeper than flash cards.
          </li>
        </ul>
        <HomePageCta />
      </section>
    </div>
  )
}

export default function HomePage() {
  /*
    if (getIsLoggedIn(data) === null) {
    return spinner()
  }
  */
  const isLoggedIn = false
  return (
    <div id="home" className="page">
      <header>
        <img
          src="/astrolabe.svg"
          className="home__logo"
          alt="astrolabe logo"
        />
        <hgroup>
          <h1>Sagefy</h1>
          <h3>Learn anything, customized for you.</h3>
          <h6>...and always free.</h6>
        </hgroup>
        {isLoggedIn ? (
          <p>
            {'Logged in. '}
            <a href="/my_subjects">
              My Subjects <Icon name="next" />
            </a>
          </p>
        ) : (
          <p>
            <a href="/log_in">
              <Icon name="log-in" /> Log In
            </a>
            {' or '}
            <a href="/sign_up">
              <Icon name="sign-up" /> Sign Up
            </a>
          </p>
        )}
      </header>
      {!isLoggedIn && <HomePageInfo />}
      <footer>
        <ul>
          <li>Â© Copyright 2018 Sagefy.</li>
          <li>
            <a href="https://docs.sagefy.org/">Docs</a>
          </li>
          <li>
            <a href="https://stories.sagefy.org/">Stories (Blog)</a>
          </li>
          <li>
            <a href="https://sgef.cc/devupdates">Updates</a>
          </li>
          <li>
            <a href="https://sgef.cc/feedback">
              <Icon name="contact" /> Support
            </a>
          </li>
          <li>
            <a href="/terms">
              <Icon name="terms" /> Privacy & Terms
            </a>
          </li>
        </ul>
      </footer>
    </div>
  )
}
