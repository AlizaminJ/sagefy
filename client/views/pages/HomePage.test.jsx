import HomePage from './HomePage'

describe('HomePage', () => {
  it('should render the home page', () => {
    expect(HomePage({ role: 'sg_anonymous' })).toMatchSnapshot()
  })

  it('should render the logged in home page', () => {
    expect(HomePage({ role: 'sg_user' })).toMatchSnapshot()
  })
})
