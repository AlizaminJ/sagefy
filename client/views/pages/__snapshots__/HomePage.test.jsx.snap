// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage should render the home page 1`] = `
<Layout
  description="What do you want to learn? Learn for free, always. Any subject you want. And Sagefy saves you time and effort by adapting to what you know and where you want to go."
  page="HomePage"
  title="Learn anything, adapted for you. Free."
>
  <section
    className="ta-c"
  >
    <h1>
      What do you want to learn? 
      <Icon
        i="search"
        s="xxl"
      />
    </h1>
    <form
      action="/search-subjects"
    >
      <p>
        <input
          aria-label="Search"
          autoFocus={true}
          name="q"
          placeholder="example: Music"
          size="40"
          type="search"
        />
      </p>
      <p>
        <button
          type="submit"
        >
          <Icon
            i="search"
            s="m"
          />
           Search
        </button>
      </p>
    </form>
  </section>
  <section
    className="ta-r"
  >
    <p>
      <small>
        <Link
          replace={false}
          to="/log-in"
        >
          <Icon
            i="logIn"
            s="m"
          />
           Log In
        </Link>
         
        or
         
        <Link
          replace={false}
          to="/sign-up"
        >
          <Icon
            i="signUp"
            s="m"
          />
           Sign Up
        </Link>
      </small>
    </p>
  </section>
  <section>
    <div
      className="ta-c m-yc"
    >
      <img
        alt="astrolabe"
        height="120"
        src="/astrolabe.svg"
      />
      <h2>
        Sagefy
      </h2>
      <p>
        <em>
          Learn anything, adapted for you. Free.
        </em>
      </p>
    </div>
  </section>
  <section>
    <ul>
      <li>
        <Icon
          i="adapt"
          s="m"
        />
         
        <strong>
          Adaptive Learning.
        </strong>
         Sagefy optimizes based on what you already know and what your goal is. Get the most out of your time and effort spent.
      </li>
      <li>
        <Icon
          i="open"
          s="m"
        />
         
        <strong>
          Open Content.
        </strong>
         Anyone can view, share, create, and edit content. Because anyone can contribute, you can learn anything you want.
      </li>
      <li>
        <ExternalLink
          href="https://youtu.be/gFn4Q9tx7Qs"
        >
          <Icon
            i="video"
            s="m"
          />
           Watch the overview video
        </ExternalLink>
        .
      </li>
    </ul>
  </section>
  <Footer
    role="sg_anonymous"
  />
</Layout>
`;

exports[`HomePage should render the logged in home page 1`] = `
<Layout
  description="What do you want to learn? Learn for free, always. Any subject you want. And Sagefy saves you time and effort by adapting to what you know and where you want to go."
  page="HomePage"
  title="Learn anything, adapted for you. Free."
>
  <section
    className="ta-c"
  >
    <h1>
      What do you want to learn? 
      <Icon
        i="search"
        s="xxl"
      />
    </h1>
    <form
      action="/search-subjects"
    >
      <p>
        <input
          aria-label="Search"
          autoFocus={true}
          name="q"
          placeholder="example: Music"
          size="40"
          type="search"
        />
      </p>
      <p>
        <button
          type="submit"
        >
          <Icon
            i="search"
            s="m"
          />
           Search
        </button>
      </p>
    </form>
  </section>
  <section
    className="ta-r"
  >
    <p>
      <small>
        <Link
          replace={false}
          to="/dashboard"
        >
          Go to 
          <Icon
            i="dashboard"
            s="m"
          />
           Dashboard
        </Link>
      </small>
    </p>
  </section>
  <section>
    <div
      className="ta-c m-yc"
    >
      <img
        alt="astrolabe"
        height="120"
        src="/astrolabe.svg"
      />
      <h2>
        Sagefy
      </h2>
      <p>
        <em>
          Learn anything, adapted for you. Free.
        </em>
      </p>
    </div>
  </section>
  <section>
    <ul>
      <li>
        <Icon
          i="adapt"
          s="m"
        />
         
        <strong>
          Adaptive Learning.
        </strong>
         Sagefy optimizes based on what you already know and what your goal is. Get the most out of your time and effort spent.
      </li>
      <li>
        <Icon
          i="open"
          s="m"
        />
         
        <strong>
          Open Content.
        </strong>
         Anyone can view, share, create, and edit content. Because anyone can contribute, you can learn anything you want.
      </li>
      <li>
        <ExternalLink
          href="https://youtu.be/gFn4Q9tx7Qs"
        >
          <Icon
            i="video"
            s="m"
          />
           Watch the overview video
        </ExternalLink>
        .
      </li>
    </ul>
  </section>
  <Footer
    role="sg_user"
  />
</Layout>
`;
